specs/Pills.md

```diff
@@ -0,0 +1,188 @@
+
+PIlls should be used when representing an input, as a way to filter content, or to represent an attribute.
+
+## Prior Art
+
+- [Evergreen](https://evergreen.segment.com/components/badge-and-pill/)
+- [Lightning](https://www.lightningdesignsystem.com/components/pills/)
+- [OpenUI Research]()
+
+## Sample Code
+
+```jsx
+<Pills>
+  <Pill>Item 1</Pill>
+  <Pill>Item 2</Pill>
+</Pills>
+```
+
+## Variants
+
+- Shape: `circular`, `rounded`
+- Size: `smaller`, `small`, `medium`
+- Appearance: `Filled Grey (filled)`, `Filled White (inverted)`, `Outline`
+
+## PROPS
+
+```typescript
+export type PillShape = 'rounded' | 'circular';
+
+export type PillAppearence = 'filled' | 'inverted' | 'outline';
+
+export type SizeValue = 'smaller' | 'small' | 'medium';
+
+export interface PillProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * A Pill can be sized.
+   */
+  size?: SizeValue;
+
+  /**
+   * A Pill can be circular or rounded
+   */
+  shape?: PillShape;
+
+  /**
+   * A Pill can be filled, inverted or outline
+   */
+  appearance?: PillVariant;
+
+  /**
+   * Media slot
+   */
+  media?: ShorthandProps<HTMLElement>;
+
+  /**
+   * Details that will appear as second line
+   */
+  details?: ShorthandProps<HTMLElement>;
+
+  /** A Pill can be dismissible. */
+  dismissible?: boolean;
+
+  /** A Pill can be disabled. */
+  disabled?: boolean;
+
+  /** A Pill can be selected. */
+  selected?: boolean;
+
+  /** A Pill can be draggable  */
+  draggable?: boolean;
+
+  /**
+   * A button shorthand for the dismiss action slot. To use this slot the pill should be
+   * dismissible.
+   */
+  dismissAction?: ShorthandValue<AlertDismissActionProps>;
+}
+```
+
+## Structure
+
+- _**Public**_
+
+```jsx
+<Pills>
+  <Pill />
+</Pills>
+```
+
+- _**Internal**_
+
+```jsx
+<ElementType >
+  <li>
+    <span>
+      {children}
+    </span>
+  <li>
+</ElementType>
+```
+
+- _**DOM**_
+
+```html
+<ul aria-label>
+  <li role="presentation">
+    <span role="button">
+      ...
+    </span>
+  </li>
+</ul>
+```
+
+## Migration
+
+- _Migration from v8_
+
+...
+
+- _Migration from v0_
+
+`Dropdown` component should use `Pills` over `DropdownSelectedItem`.
+
+### Pill with dismiss
+
+#### Accessibility
+
+- Each pill is of role="button"
+- Tab key to set focus on the Pill
+- Tab key moves focus to next item
+- Esc key remove focus
+
+#### Keyboard Navigation
+
+| key                | state    | title                                                              |
+| ------------------ | -------- | ------------------------------------------------------------------ |
+| left/right         | OnFocus  | Role= “button”, aria-describedby=”Press enter or delete to remove” |
+| Enter/Delete       | OnDelete | The pill is removed since there is only one primary action         |
+| Backspace/Ctrl + X | OnDelete | Backspace/Ctrl + X                                                 |
+
+### Pill with popup
+
+#### Accessibility
+
+- Each pill is of role="button"
```

Comment:
if all pills are of `role="button"` we should state that earlier instead of requiring the reader to check each variation